File: expensemanagement.web/src/api/surplusAllocation.ts
import api from "@/service/axios";

export const allocateSurplusApi = async (dto: {
  savingsGoalId: number;
  month: number;
  year: number;
  amount: number;
}) => {
  const res = await api.post("/surplusallocation", dto);
  return res.data;
};

File: ExpenseManagement.API/ExpenseManagement.API/Models/SurplusAllocation.cs
namespace ExpenseManagement.API.Models
{
  
        public class SurplusAllocation
        {
            public int Id { get; set; }
            public int Month { get; set; }
            public int Year { get; set; }
            public decimal Amount { get; set; }
            public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

            public int SavingsGoalId { get; set; }
            public SavingsGoal SavingsGoal { get; set; } = null!;

            public string UserId { get; set; } = string.Empty;
            public ApplicationUser User { get; set; } = null!;
        }
}
